core/src/main/kotlin/behavior/GuildBehavior.kt

```diff
@@ -192,7 +192,7 @@ interface GuildBehavior : KordEntity, Strategizable {
      */
     val gateway: Gateway?
         get() {
-            val shard = id.value.shr(22) % kord.resources.shardCount.coerceAtLeast(1)
+            val shard = id.value.shr(22) % kord.resources.shards.indices.count().coerceAtLeast(1)
```

Comment:
Shouldn't we be calculating the shard based on the mod by totalShards instead? 
```suggestion
            val shard = id.value.shr(22) % kord.resources.shards.totalShards.coerceAtLeast(1)
```

Reply:
Good catch